### PR TITLE
Add `statistics-graph-card`

### PR DESCRIPTION
**Pull request by bramkragten**
## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/components/chart/statistics-chart.ts [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+    statisticsData.forEach((stats) => {
+      const firstStat = stats[0];
+      let name = names[firstStat.statistic_id];
+      if (!name) {
+        const entityState = this.hass.states[firstStat.statistic_id];
+        if (entityState) {
+          name = computeStateName(entityState);
+        } else {
+          name = firstStat.statistic_id;
+        }
```
Comment (ludeeus) [inline src/components/chart/statistics-chart.ts:202]:
```suggestion
        name = entityState ? computeStateName(entityState) : firstStat.statistic_id;
```
Alternative :man_shrugging:

File: src/components/chart/statistics-chart.ts [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+      const pushData = (timestamp: Date, datavalues: any[] | null) => {
+        if (!datavalues) return;
+        if (timestamp > endTime) {
+          // Drop datapoints that are after the requested endTime. This could happen if
+          // endTime is "now" and client time is not in sync with server time.
+          return;
+        }
+        data.forEach((d, i) => {
+          if (datavalues[i] === null && prevValues && prevValues[i] !== null) {
+            // null data values show up as gaps in the chart.
```
Comment (ludeeus) [inline src/components/chart/statistics-chart.ts:218]: Can we traverse the data to get the first non-null value instead?

File: src/components/chart/statistics-chart.ts [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+      let lastDate: Date;
+
+      // Process chart data.
+      stats.forEach((stat) => {
+        const value: Array<number | null> = [];
+        statTypes.forEach((type) => {
+          value.push(
+            stat[type] ? Math.round(stat[type]! * 100) / 100 : stat[type]
+          );
+        });
```
Comment (ludeeus) [inline src/components/chart/statistics-chart.ts:275]:
```suggestion
        statTypes.forEach((type) => {
          if (stat[type]) {
            value.push(Math.round(stat[type] * 100) / 100);
          }
        });
```
There is no need to push these right?

**Review by ludeeus — DISMISSED** [review]

File: src/components/chart/statistics-chart.ts (hunk 3 above)
Comment (bramkragten) [reply src/components/chart/statistics-chart.ts:271]: We should push `null` when the data is not available.

File: src/components/chart/statistics-chart.ts (hunk 2 above)
Comment (bramkragten) [reply src/components/chart/statistics-chart.ts:214]: That is not needed? The first time a `null` value gets in `prevValue` can't be `null`, except when it is the start of the graph, and then we don't care.

**Pull request merged**